qark/decompiler/decompiler.py

```diff
@@ -46,25 +46,47 @@ def __init__(self, path_to_apk, build_directory=None):
         """
         self.path_to_apk = path_to_apk
         self.build_directory = build_directory if build_directory else os.path.join(os.path.dirname(path_to_apk), "qark")
+
+        # validate we are running on an APK, Directory, or Java source code
+        if not os.path.exists(self.path_to_apk):
+            return
+
+        if os.path.isfile(self.path_to_apk) and os.path.splitext(self.path_to_apk.lower())[1] not in (".java", ".apk"):
```

Comment:
`path_to_apk` doesn't seem appropriate anymore. You should refactor this out now, including maybe changing the method signature, before it becomes a bigger pain later.